pkg/kubelet/kubelet_test.go

```diff
@@ -1610,6 +1610,119 @@ func TestFilterOutInactivePods(t *testing.T) {
 	assert.Equal(t, expected, actual)
 }
 
+func TestHandleMaxContainerCheckpoints(t *testing.T) {
+	testKubelet := newTestKubelet(t, false /* controllerAttachDetachEnabled */)
+	defer testKubelet.Cleanup()
+	kubelet := testKubelet.kubelet
+
+	// For the test let's set it to the smallest value.
+	kubelet.maxContainerCheckpoints = 1
+	// The checkpoint directory creation is behind the feature gate.
+	// Let's create it manually here.
+	// It will be removed by the Cleanup() function.
+	require.NoError(t, os.MkdirAll(kubelet.getCheckpointsDir(), 0o700))
+
+	containerID := kubecontainer.ContainerID{
+		Type: "test",
+		ID:   "abc1234",
+	}
+
+	fakePod := &containertest.FakePod{
+		Pod: &kubecontainer.Pod{
+			ID:        "12345678",
+			Name:      "podFoo",
+			Namespace: "nsFoo",
+			Containers: []*kubecontainer.Container{
+				{
+					Name: "containerFoo",
+					ID:   containerID,
+				},
+			},
+		},
+	}
+
+	podFullName := fmt.Sprintf(
+		"%s_%s",
+		fakePod.Pod.Name,
+		fakePod.Pod.Namespace,
+	)
+	containerName := fakePod.Pod.Containers[0].Name
+
+	options := &runtimeapi.CheckpointContainerRequest{}
+	// First checkpoint archive
+	firstCheckpointArchive := filepath.Join(
+		kubelet.getCheckpointsDir(),
+		fmt.Sprintf(
+			"checkpoint-%s-%s-2023-04-13T12:13:24Z.tar",
+			podFullName,
+			containerName,
+		),
+	)
+
+	options.Location = firstCheckpointArchive
+
+	checkpointContent := "checkpoint archive"
+	err := os.WriteFile(options.Location, []byte(checkpointContent), 0o600)
+	require.NoError(t, err)
+
+	err = testKubelet.kubelet.handleMaxContainerCheckpoints(
+		fakePod.Pod.ID,
+		podFullName,
+		containerName,
+		options,
+	)
+	require.NoError(t, err)
+	// The first checkpoint archive should still exist
+	_, err = os.Stat(firstCheckpointArchive)
+	require.NoError(t, err)
+
+	time.Sleep(1 * time.Second)
+	// second checkpoint archive
+	secondCheckpointArchive := filepath.Join(
+		kubelet.getCheckpointsDir(),
+		fmt.Sprintf(
+			"checkpoint-%s-%s-2023-04-13T12:13:25-08:00.tar",
+			podFullName,
+			containerName,
+		),
+	)
+
+	options.Location = secondCheckpointArchive
+
+	err = os.WriteFile(options.Location, []byte(checkpointContent), 0o600)
+	require.NoError(t, err)
```

Comment:
Would it make sense to wrap test file creation into a function? This can make a test code shorter and more readable.